### PR TITLE
Added tests for store API V2.0.0

### PR DESCRIPTION
**Pull request by mirceaalexandru**

### CODE REVIEW

**Comment by mirceaalexandru** [comment]
@paolochiodi 

API V2.0.0 will contain support for:

ne$: .list$({ f1: {ne$: v1} }) for not-equal.
eq$: .list$({ f1: {eq$: v1} }) for equal.
lte$: .list$({ f1: {lte$: 5} }) for less than or equal.
lt$: .list$({ f1: {lt$: 5} }) for less than.
gte$: .list$({ f1: {gte$: 5} }) for greater than or equal.
gt$: .list$({ f1: {gt$: 5} }) for greater than.
in$: .list$({ f1: {in$: [10, 20]} }) for in. in$ operator accepts only values of type array.
nin$: .list$({ f1: {nin$: ['v1', 'v2']} }) for not-in. nin$ operator accepts only values of type array.

and also logical operators:

or$: .list$({ or$: [{name: 'something'}, {price: 200}]})
and$: .list$({ and$: [{name: 'something'}, {price: 200}]})


Please see latest version of seneca-postgres-store.

**Comment by paolochiodi** [comment]
Looks like I'm out of the seneca loop.
Are the new api approved by @rjrodger? 
Last time I checked he wanted to keep the entity framework quite simple and wasn't looking to add "complex" conditions.

**Comment by rjrodger** [comment]
Hi guys - the extended api is optional, so tests for it should use a flag to trigger them

**Comment by mirceaalexandru** [comment]
Yes, this new API was implemented in seneca-postgres as the current set of features where not enough for a project. These new features seems to be required and it was discussed with @rjrodger  - don't know  right now the discussion place. We can define API 2.0 and one by one change stores to match this list.

We can specify in each store what API is supported.

**Comment by mirceaalexandru** [comment]
Oh, I was written my comment and did not see @rjrodger 's comment.

**Comment by mirceaalexandru** [comment]
@paolochiodi this is now passing as I fixed according with latest linting rules.

**Comment by rjrodger** [comment]
We will need to document this 2 layer entity API.

**Comment by mirceaalexandru** [comment]
@rjrodger My proposal was to have V1 and V2 documented somewhere - maybe in a section in senecajs.org site? Then all stores can refer to that as API description, with clear notes if something from that API cannot be implemented due to limited DB features.

This is required also in order to make sure that we do not write same API description in all stores documentation.

We can have even some badges/mark/something in the store readme to show clearly that a store fully complies with a specific API and that is "SenecaJS approved"?

**Comment by mcdonnelldean** [comment]
@mirceaalexandru @rjrodger Related, https://github.com/senecajs/senecajs.org/issues/106

**Comment by mirceaalexandru** [comment]
@mcdonnelldean maybe this can be done as lodash for example: with a selector for API documentation.
https://lodash.com/docs
Senecajs will remain a site with general description, tutorials, list of plugins with their general descriptions and so on. Separately we can have API documentation for each plugin, with version selector.

**Comment by mcdonnelldean** [comment]
@mirceaalexandru sure, makes sense.

File: package.json [hunk 1]
```diff
@@ -34,7 +35,7 @@
     "lodash": "^3.10.1"
   },
   "peerDependencies": {
-    "seneca": ">=0.8.0 <0.9.0"
+    "seneca": ">=0.8.0"
```
Comment (paolochiodi) [inline package.json]: Not sure If I like this. What's the reason for not having an upper bound?

**Comment by paolochiodi** [comment]
Hi,
sorry for coming back so late to this PR.
Overall I'm happy whit it, just a couple of indications
- I'd like to change the name of the new set of tests. Something like "rich-queries" or "advanced-filters" in stead of v2
- I'd like to have two separate PRs here, one for the actual v2 and one for the other changes (readme, upper case etc)

**Comment by mihaidma** [comment]
@paolochiodi @mirceaalexandru I need this PR please to be unblocked and moved forward.

To unblock it I propose the following:
- Paolo: let's not split it into two PRs and merge it as it is,
- Mircea: fix the conflicts and rename the test,
- Let's clarify asap the naming/versioning.

**My naming proposal:**

For the current basic layer of tests we have 4 categories of tests:
`basictest, sorttest, limitstest, sqltest`
I propose to name the 2-nd layer `extendedtest`. That should be implemented by sql stores and mongo.

Then if need arises we could version the layers. At this moment I think all stores should comply with latest version of tests (only for the layers supported by DB type), no point to version them.
In time we should better name the layers and define what they are supposed to do.
I suppose there is a reason for which there are 4 basic layers of tests else we'll merge them in time.

Each DB store maintainer explicitly calls in his tests the layers of tests specific to that DB type.

**Comment by paolochiodi** [comment]
I don't like `extendedtest`too much - I'd like something more specific. But if that's ok for you and @mirceaalexandru I'll go ahead and merge the PR

**Comment by mihaidma** [comment]
I'll put for the reference the layers defined by Richard in a mail:
- key-value
- basic queries
- extended queries (like mongodb)
- knex for relational models
- massivejs for a full feature ORM

**Comment by mihaidma** [comment]
@paolochiodi the PR is fixed. Could you please review and merge? It would be good if we also publish a new version of tests. I can help you if you need it

**Pull request merged**